INFO     fixtures.settings_fixtures:settings_fixtures.py:18 Attempting to load settings for test session
DEBUG    config.settings:settings.py:45 Timeout value 10.0 validated successfully
DEBUG    config.settings:settings.py:55 X-Challenger token validated successfully
DEBUG    config.settings:settings.py:23 Settings initialized: {'api_url': HttpUrl('https://apichallenges.eviltester.com/'), 'api_timeout': 10.0, 'api_x_challenger': '0ec06db1-e4df-46f7-87cf-6b414d0d3d36', 'log_level': 'DEBUG'}
INFO     config.settings:settings.py:24 Settings class successfully instantiated
DEBUG    fixtures.settings_fixtures:settings_fixtures.py:21 Settings loaded successfully: {'api_url': HttpUrl('https://apichallenges.eviltester.com/'), 'api_timeout': 10.0, 'api_x_challenger': '0ec06db1-e4df-46f7-87cf-6b414d0d3d36', 'log_level': 'DEBUG'}
INFO     fixtures.settings_fixtures:settings_fixtures.py:22 Settings fixture initialized for test session
INFO     fixtures.clients_fixtures:clients_fixtures.py:19 Creating HTTP client for test session
DEBUG    fixtures.clients_fixtures:clients_fixtures.py:25 HTTP client created with base URL: https://apichallenges.eviltester.com/ and timeout: 10.0
INFO     fixtures.clients_fixtures:clients_fixtures.py:44 Creating ChallengesClient for test session
INFO     clients.base_client:base_client.py:24 BaseClient initialized with provided HTTP client
INFO     clients.challenges_client:challenges_client.py:22 ChallengesClient initialized with endpoint: /challenges
DEBUG    fixtures.clients_fixtures:clients_fixtures.py:47 ChallengesClient initialized successfully
INFO     clients.challenges_client:challenges_client.py:33 Fetching all challenges from API
DEBUG    clients.challenges_client:challenges_client.py:34 Request parameters - headers: None, params: None
DEBUG    clients.base_client:base_client.py:71 Preparing GET request to /challenges with params None
INFO     clients.base_client:base_client.py:52 Make GET request to /challenges with headers {'accept': '*/*', 'accept-encoding': 'gzip, deflate', 'connection': 'keep-alive', 'user-agent': 'python-httpx/0.28.1', 'x-challenger': '0ec06db1-e4df-46f7-87cf-6b414d0d3d36'} params None
INFO     tools.attachments:attachments.py:19 Attaching request data to Allure for GET /challenges
DEBUG    tools.attachments:attachments.py:24 Attaching request headers: {'accept': '*/*', 'accept-encoding': 'gzip, deflate', 'connection': 'keep-alive', 'user-agent': 'python-httpx/0.28.1', 'x-challenger': '0ec06db1-e4df-46f7-87cf-6b414d0d3d36'}
INFO     tools.attachments:attachments.py:80 Request data successfully attached to Allure
INFO     clients.base_client:base_client.py:58 Received response with status 200 from /challenges
INFO     tools.attachments:attachments.py:148 Attaching response data to Allure for GET /challenges
DEBUG    tools.attachments:attachments.py:158 Attaching response headers: {'access-control-allow-credentials': 'true', 'access-control-allow-headers': '*', 'access-control-allow-methods': '*', 'access-control-allow-origin': '*', 'access-control-expose-headers': '*', 'content-type': 'application/json', 'date': 'Sat, 31 May 2025 19:02:24 GMT', 'location': '/gui/challenges/0ec06db1-e4df-46f7-87cf-6b414d0d3d36', 'nel': '{"report_to":"heroku-nel","response_headers":["Via"],"max_age":3600,"success_fraction":0.01,"failure_fraction":0.1}', 'report-to': '{"group":"heroku-nel","endpoints":[{"url":"https://nel.heroku.com/reports?s=iG2TvxjkeF%2B4VUC5lN8HgtAoW40xk2gTpRonZKbeWwk%3D\\u0026sid=c46efe9b-d3d2-4a0c-8c76-bfafa16c5add\\u0026ts=1748718144"}],"max_age":3600}', 'reporting-endpoints': 'heroku-nel="https://nel.heroku.com/reports?s=iG2TvxjkeF%2B4VUC5lN8HgtAoW40xk2gTpRonZKbeWwk%3D&sid=c46efe9b-d3d2-4a0c-8c76-bfafa16c5add&ts=1748718144"', 'server': 'Heroku', 'via': '1.1 heroku-router', 'x-challenger': '0ec06db1-e4df-46f7-87cf-6b414d0d3d36', 'x-robots-tag': 'noindex', 'transfer-encoding': 'chunked'}
DEBUG    tools.attachments:attachments.py:168 Processing response body with content type: application/json
DEBUG    tools.attachments:attachments.py:175 JSON response body formatted successfully
INFO     tools.attachments:attachments.py:208 Response data successfully attached to Allure
INFO     clients.challenges_client:challenges_client.py:38 Successfully fetched challenges, status code: 200
INFO     tools.assertions.base_assertions:base_assertions.py:448 Starting comprehensive checks for API response
DEBUG    tools.assertions.base_assertions:base_assertions.py:449 Checks to perform: {'status_code': <HTTPStatus.OK: 200>, 'response_time': 5.0, 'schema': True, 'headers_present': ['content-type'], 'header_values': None, 'key_present': ['challenges'], 'key_value': {'id': 59}, 'request_headers': None}
INFO     tools.assertions.base_assertions:base_assertions.py:31 Checking status code, expected: 200
DEBUG    tools.assertions.base_assertions:base_assertions.py:41 Status code check passed: 200
INFO     tools.assertions.base_assertions:base_assertions.py:52 Checking response time, max allowed: 5.0s
DEBUG    tools.assertions.base_assertions:base_assertions.py:67 Response time check passed: 0.105s
INFO     tools.assertions.base_assertions:base_assertions.py:78 Checking for presence of headers: ['content-type']
DEBUG    tools.assertions.base_assertions:base_assertions.py:88 Header presence check passed for: ['content-type']
INFO     tools.assertions.base_assertions:base_assertions.py:99 Checking header values: {'content-type': 'application/json'}
DEBUG    tools.assertions.base_assertions:base_assertions.py:113 Header values check passed for: {'content-type': 'application/json'}
INFO     tools.assertions.base_assertions:base_assertions.py:215 Validating JSON schema: ChallengesSchema
DEBUG    tools.assertions.base_assertions:base_assertions.py:229 JSON schema validation passed for: ChallengesSchema
INFO     tools.assertions.base_assertions:base_assertions.py:124 Checking for presence of body keys: ['challenges']
DEBUG    tools.assertions.base_assertions:base_assertions.py:140 Body keys presence check passed for: ['challenges']
INFO     tools.assertions.base_assertions:base_assertions.py:151 Checking body key values: {'id': 59}
DEBUG    tools.assertions.base_assertions:base_assertions.py:204 Body key values check passed for: {'id': 59}
INFO     tools.assertions.base_assertions:base_assertions.py:488 Comprehensive checks completed successfully